akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -790,10 +790,33 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * '''Cancels when''' downstream cancels
    *
    */
+  @deprecated("Use recoverWithRetries instead.", "2.4.3")
   def recoverWith[T >: Out](pf: PartialFunction[Throwable, _ <: Graph[SourceShape[T], NotUsed]]): javadsl.Flow[In, T, Mat @uncheckedVariance] =
```

Comment:
I'm not convinced about that, what about https://github.com/akka/akka/pull/20137#issuecomment-203833534